manifests/ipa/install.pp

```diff
@@ -0,0 +1,121 @@
+# Run ipa-client-install on puppet clients
+#
+# @param ensure
+#   'present' to install, 'absent' to uninstall
+#
+# @param enroll
+#   Be smart about when to run `ipa-client-install`
+#
+# @param ip_address
+#   IP address of host being connected
+#
+# @param hostname
+#   Hostname of the host being connected
+#
+# @param password
+#   Admin or host-based one-time-password generated by `ipa host-add` or the GUI
+#
+# @param server
+#   IPA server to connect to
+#
+# @param domain
+#   IPA Domain
+#
+# @param realm
+#   IPA Realm
+#
+# @param no_ac
+#   Run without authconfig
```

Comment:
authconfig is not enabled in a SIMP system.  In fact, the authconfig executable is either removed from the system (old SIMP) or linked to /bin/true.  So, exposing this only makes sense for non-SIMP systems.

Reply:
I don't want to obscure any functionality of `ipa-client-install`, and this module might be run on systems where authconfig is used as intended. It still defaults to true.